app/models/tenant.rb

```diff
@@ -30,7 +32,13 @@ def self.resolve_host(host)
       host_domain = allowed_domains.find { |domain| host == domain || host.ends_with?(".#{domain}") }
       schema = host_without_www.sub(/\.?#{host_domain}\Z/, "").presence
 
-      schema unless find_by_domain(schema)
+      if schema
+        if only_hidden.or(domain).find_by(schema: schema)
+          raise Apartment::TenantNotFound
+        else
+          schema
+        end
+      end
```

Comment:
I wonder if we could reduce conditional nesting within this method.

```suggestion
      raise Apartment::TenantNotFound if schema && only_hidden.or(domain).find_by(schema: schema)
      
      schema
```

Reply:
I've updated the code.